config/crds/resources/apiextensions.k8s.io_v1_customresourcedefinition_computenetworkfirewallpolicyrules.compute.cnrm.cloud.google.com.yaml

```diff
@@ -0,0 +1,362 @@
+                    name:
+                      description: Name of the secure tag, created with TagManager's
+                        TagValue API. @pattern tagValues/[0-9]+.
+                      type: string
+                    state:
+                      description: '[Output Only] State of the secure tag, either
+                        `EFFECTIVE` or `INEFFECTIVE`. A secure tag is `INEFFECTIVE`
+                        when it is deleted or its network is deleted.'
+                      type: string
+                  required:
+                  - name
+                  type: object
+                type: array
+              targetServiceAccounts:
+                description: A list of service accounts indicating the sets of instances
+                  that are applied with this rule.
+                items:
+                  type: string
+                type: array
+            required:
+            - action
+            - direction
+            - firewallPolicyRef
+            - match
+            - projectRef
+            type: object
+          status:
+            properties:
+              conditions:
+                description: Conditions represent the latest available observation
+                  of the resource's current state.
+                items:
+                  properties:
+                    lastTransitionTime:
+                      description: Last time the condition transitioned from one status
+                        to another.
+                      type: string
+                    message:
+                      description: Human-readable message indicating details about
+                        last transition.
+                      type: string
+                    reason:
+                      description: Unique, one-word, CamelCase reason for the condition's
+                        last transition.
+                      type: string
+                    status:
+                      description: Status is the status of the condition. Can be True,
+                        False, Unknown.
+                      type: string
+                    type:
+                      description: Type is the type of the condition.
+                      type: string
+                  type: object
+                type: array
+              kind:
+                description: Type of the resource. Always `compute#firewallPolicyRule`
+                  for firewall policy rules.
+                type: string
+              observedGeneration:
+                description: ObservedGeneration is the generation of the resource
+                  that was most recently observed by the Config Connector controller.
+                  If this is equal to metadata.generation, then that means that the
+                  current reported status reflects the most recent desired state of
+                  the resource.
+                type: integer
+              ruleTupleCount:
```

Comment:
We have a design decision that GCP resource specific fields should be moved from `status` to `observedState`: #1296
Though the implementation didn't cover the manually configured v1alpha1 resources: #1353
I'll need to add the support for this resource.

I personally don't think that's a blocker because this is a v1alpha1 resource with a specific consumer (the self-service contributor themselves) but @justinsb I'll let you make the decision. Context about why it is v1alpha1: b/315045017#comment9 and on.

Reply:
@katrielt how much can your team tolerate breaking changes in the schema? 

I have a WIP change #1658 to address the schema for v1alpha1 CRDs though it might take sometime before I fix all the unit test issues there. We can either merge the resource as is and update the schema later; or wait for #1658.

Reply:
@maqiuyujoyce will the change in #1658 mean we have to make changes to the resource definition? If that's the case, what happens to the existing, defined, resource?

Generally, I would rather wait after the implementation to minimize changes over time, but if this change doesn't require later changes to the resource definitions, then let's merge.

Reply:
#1658 creates breaking changes to the CRD; they are the same breaking changes if we finally graduate this resource to v1beta1.

For any breaking changes in the CRD, we suggest users [abandon](https://cloud.google.com/config-connector/docs/how-to/managing-deleting-resources#keeping_resources_after_deletion) the existing resources and [acquire](https://cloud.google.com/config-connector/docs/how-to/managing-deleting-resources#acquiring_an_existing_resource) them; otherwise the reconciliation may hit issues.

To minimize the maintenance work in the future, I think it's probably better to make this PR blocked on #1658.

Reply:
@maqiuyujoyce If that's the case, then lets move forward with merging this PR.

I'll explain - Rules are attached to a Policy, which is in turn attached to a VPC. It would be relatively easy to  create a new policy and set of matching rules and update the VPC attachement. The old policy and rules can just be deleted after the policy is switched.

The question that remains is how would I know when I need to do this change?

Reply:
SGTM. Thank you!